os/board/common/crashdump.c

```diff
@@ -238,6 +317,63 @@ static int send_fsdump(void)
 	return 0;
 }
 
+static int send_extfsdump(void)
+{
+	int i = 0;
+	int fd;
+	int index;
+	uint8_t *ptr;
+	size_t size;
+	char ext_userfs_partition[CONFIG_PATH_MAX] = { '\0' };
+	char ext_fsdump_buf[CONFIG_MTD_SMART_SECTOR_SIZE + 1];
+
+	/*Send ACK for EXTERNAL FSDUMP Handshake */
+	up_lowputc('A');
+
+	i = get_ext_userfs_part(&size, &index);
+	if (i < 0) {
+		return ERROR;
+	}
+
+	/* Send external userfs partition size to host */
+	ptr = (uint8_t *)&size;
+	for (i = 0; i < sizeof(size); i++) {
+		up_lowputc((uint8_t)*ptr);
+		ptr++;
+	}
+
+	/* If extracted size is 0, it means there is no partition available to dump */
+	if (size == 0) {
+		return 0;
+	}
+
+	snprintf(ext_userfs_partition, CONFIG_PATH_MAX, "/dev/smart1p%d", index);
```

Comment:
hmm...it should be handled by config value.
Let's add TOTO here and fix it when you have some times.

Reply:
Updated